lucille-core/src/test/java/com/kmwllc/lucille/stage/ChunkingTest.java

```diff
@@ -0,0 +1,379 @@
+    Document child3 = childrenDocs.get(2);
+
+    assertEquals("The sun is bright.", child1.getString("chunk"));
+    assertEquals("The sky is clear.", child2.getString("chunk"));
+    assertEquals("Birds are singing.", child3.getString("chunk"));
+  }
+
+  @Test
+  public void testParagraphChunking() throws StageException {
+    Stage stage = factory.get("ChunkingTest/paragraphChunk.conf");
+    stage.start();
+
+    Document doc = Document.create("id");
+    doc.setField("text", "This is the first\nparagraph of the document.\n\n This is the second paragraph of the document."
+        + "\n \n This is the third paragraph of the document. \n \n This is the fourth paragraph of the document.");
+
+    stage.processDocument(doc);
+
+    List<Document> childrenDocs = doc.getChildren();
+    Document child1 = childrenDocs.get(0);
+    Document child2 = childrenDocs.get(1);
+    Document child3 = childrenDocs.get(2);
+    Document child4 = childrenDocs.get(3);
+
+    assertEquals("This is the first paragraph of the document.", child1.getString("chunk"));
+    assertEquals("This is the second paragraph of the document.", child2.getString("chunk"));
+    assertEquals("This is the third paragraph of the document.", child3.getString("chunk"));
+    assertEquals("This is the fourth paragraph of the document.", child4.getString("chunk"));
+  }
+
+
+
+  @Test
+  public void testParagraphChunkingWithLimits() throws StageException {
+    Stage stage = factory.get("ChunkingTest/paragraphChunk.conf");
+    stage.start();
+
+    Document doc = Document.create("id");
+    doc.setField("text", "This is a very long first paragraph in document. This second sentence exceeds the character limit.\n\n This is the second paragraph of the document");
+
+    stage.processDocument(doc);
+
+    List<Document> childrenDocs = doc.getChildren();
+    Document child1 = childrenDocs.get(0);
+    Document child2 = childrenDocs.get(1);
+
+    assertEquals("This is a very long first paragraph in document. This second", child1.getString("chunk"));
+    assertEquals("This is the second paragraph of the document", child2.getString("chunk"));
+  }
+
+  @Test
+  public void testFixedChunking() throws StageException {
+    Stage stage = factory.get("ChunkingTest/fixedChunk.conf");
+    stage.start();
+
+    Document doc = Document.create("id");
+    doc.setField("text", "This is the\nfirst paragraph.\n\nThis is the second paragraph");
+
+    stage.processDocument(doc);
+
+    List<Document> childrenDocs = doc.getChildren();
+    Document child1 = childrenDocs.get(0);
+    Document child2 = childrenDocs.get(1);
+    Document child3 = childrenDocs.get(2);
+    Document child4 = childrenDocs.get(3);
+
+    assertEquals("This is the fir", child1.getString("chunk"));
+    assertEquals("st paragraph.  ", child2.getString("chunk"));
+    assertEquals("This is the sec", child3.getString("chunk"));
+    assertEquals("ond paragraph", child4.getString("chunk"));
```

Comment:
Would be good to assert that the returned string has the correct number of characters for at least one of these.